### DISCUSSION TITLE
Setting a static network/ip for gluetun and dependant containers?

### DISCUSSION THREAD

**Discussion by begunfx** [qdm12/gluetun#2328] [Q&A]
Question for everyone.  Currently, when deploying gluetun it appears to have a dynamic network and IP.   I have several dependant containers and all have healthchecks to restarts automatically should gluetun need to restart due to an unhealthy state.  That seems to work okay.  However, if I stop/restart gluetun manually then the dependent containers won't restart.  Additionally I have to restart them each 2x.  The first time I get an error and the second time they restart successfully.  What I suspect is happening is Gluetun's IP address is changing and the dependent containers are still looking for the old IP address so that's why they fail (the first try).  So, my solution was to possibly setup gluetun with a static network/IP address so that it never changes, thinking this won't break the network relationship between dependant containers.  What are your thoughts on this and would this solve the restart problem?

Thanks in advance for your consideration.

**Comment by epd5** [comment]
I'm not 100% if defining the network & addresses alone will resolve your restart issue but its worth quickly experimenting.

A little more complex but in Docker Compose I've had success with these:

Defining a network:

>      internal:
>           driver: bridge
>           name: internal
>           ipam:
>                driver: default
>                config:
>                     - subnet: 172.20.0.0/24
>                       gateway: 172.20.0.1

Then placing this under Gluetun:

>           networks:
>                internal:
>                     ipv4_address: 172.20.0.100

For the Gluetun healthcheck:

>           healthcheck:
>                test: ["CMD-SHELL", "(wget -qO- http://portcheck.transmissionbt.com/YOURPORTHERE | grep -q 1) && (wget -qO- http://portcheck.transmissionbt.com/YOURPORTHERE | grep -q 1) || exit 1"]
>                interval: 300s
>                timeout: 30s
>                retries: 5
>                start_period: 30s

I also use autoheal to restart unhealthy containers:

>      autoheal:
>           image: willfarrell/autoheal:latest
>           container_name: autoheal
>           network_mode: none
>           environment:
>                - AUTOHEAL_CONTAINER_LABEL=autoheal
>                - AUTOHEAL_INTERVAL=120
>                - AUTOHEAL_START_PERIOD=120
>                - AUTOHEAL_DEFAULT_STOP_TIMEOUT=10
>                - AUTOHEAL_ONLY_MONITOR_RUNNING=false
>                - DOCKER_SOCK=/var/run/docker.sock
>                - CURL_TIMEOUT=30
>           volumes:
>                - /var/run/docker.sock:/var/run/docker.sock
>                - /etc/localtime:/etc/localtime:ro
>           restart: unless-stopped

Then for the healthchecks for the containers dependent on Gluetuns control server responding

>           healthcheck:
>                test: ["CMD-SHELL", "curl -fs -o /dev/null http://localhost:8000/v1/openvpn/status  || exit 1"]
>                interval: 30s
>                timeout: 30s
>                retries: 3
>                start_period: 10s

I've found this reliable (if a bit slow, timings definitely need optimization) for when Gluetun reports unhealthy. 
Keen to hear any corrections/improvements/optimizations.

**Comment by kdougherty6767** [comment]
Did you ever come up with a solution to this? I am still having trouble to this day .
